src/inference/src/compilation_context.hpp

```diff
@@ -20,6 +20,10 @@ struct NetworkCompilationContext final {
 
     static std::string computeHash(const std::string& modelName,
                                    const std::map<std::string, std::string>& compileOptions);
+    static std::string computeHash(const std::string& modeStr,
+                                   const char* dataPtr,
+                                   const size_t dataCount,
```

Comment:
Can we use `ov::Tensor` instead of these arguments?

Reply:
yes, but it will lead to `ov::CoreImpl` to be friend class of `ov::Tensor`, does it make sense?

Reply:
It seems that current ov::CoreImpl uses `ie::Blob::CPtr` rather than `ov::Tensor`, it will cannot get `Blob::CPtr` from `ov::Tensor` in  ov::CoreImpl namespace, we can have 2 ways to resolve it:
1. Define a public method in ov::Tensor to get its `std::shared_ptr<InferenceEngine::Blob> _impl`
or 2. Refactor ov::CoreImpl make it use `ov::Tensor` to replace `ie::Blob::CPtr`

@ilyachur how to you think if we use ov::Tensor to compute hash?

Reply:
yes, it is a solution but there will be a little trick - seem like a short term remedy.

Reply:
done